SeaPublicWebsite/Services/RecommendationService.cs

```diff
@@ -180,49 +181,48 @@ public static List<Recommendation> GetRecommendationsForUser(UserDataModel userD
         {
             BreRequest request = CreateRequest(userData);
 
-            string requestString = JsonConvert.SerializeObject(request);
-            return BreApi.GetRecommendationsForUserRequest(requestString);
+            return BreApi.GetRecommendationsForUserRequest(request).Result;
```

Comment:
Please don't use `.Result` with async method unless you really understand async stuff. This is a potential deadlock again.
Once a method becomes async then generally everything above it in the call stack needs to become async too (which can be a pain).

Reply:
Ah okay, I've awaited (and suffixed with Async) the following: BreApi.GetRecommendationsForUserRequestAsync, RecommendationService.GetRecomendationForUserAsync, EnergyEfficiencyController.YourRecommendations_GetAsync.